### DISCUSSION TITLE
Saving event data together with detector pixel positions

### DISCUSSION THREAD

**Discussion by fott** [scipp/scipp#2143] [General]
Dear developers,

We are currently trying to use scipp to handle data from a new TOF spectrometer on a small neutron source (DIoGENE@IPHI)

We managed to create <data> which contains the event data.
However it appears to be possible to add the pixel position data (variable <position>) only once a binning operation has been performed (variable <data_binned>)
At this stage we are in a situation similar to the POWGEN example.
The <data_binned> can then be used to nicely process the data (while still containing the low level event data structure)

However, we also expected to eventually use scipp to store the data in a long term format (HDF type) (and bypassing Mantid legacy.)

The export to hdf5 is working nicely except that when saving the dataset <data> we cannot save the vector <position>.
When saving <data_binned>, while the pixel position vector is saved, only the binned data are saved and the event structure is lost.

Hence our question, how is it possible to save both the event data and the detector pixel positions in a single file ?

Best Regards

Attached, a notebook illustrating the way we are creating <data> and <data_binned>
[DIoGENE_test2.zip](https://github.com/scipp/scipp/files/7092363/DIoGENE_test2.zip)

**Comment by nvaytet** [comment]
Hi there, first of all, apologies about the delay on our response.
For some reason, it seems that none of us received a notification about your question. It seems Discussions on github is still quite new.
The notification system does appear to be working now, so you can continue to use this platform in the future.

It's great to see people using scipp.
We will look into your request and get back to you asap (over the next few days).

**Comment by SimonHeybrock** [comment]
@fott Saving of vectors should not cause any trouble, it should be supported in scipp 0.5.0 or later. Which version are you using?

```python
data_binned.to_hdf5(filename='test.h5')
```
should save events as well as coordinates (such as the position vectors you added).

Here is a working example, showing that saving event data and pixel positions works out of the box (requires a recent [scippneutron](https://scipp.github.io/scippneutron/getting-started/installation.html#with-the-provided-environment-file)):
```python
import scippneutron as scn
import scipp as sc
da = scn.data.powder_sample()
da.to_hdf5('event-data.h5')
sc.io.open_hdf5('event-data.h5')
```
